frontend/public/components/machine-config.tsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as React from 'react';
+
+import { MachineConfigKind, referenceForModel } from '../module/k8s';
+import { MachineConfigModel } from '../models';
+import { fromNow } from './utils/datetime';
+import {
+  ColHead,
+  DetailsPage,
+  List,
+  ListHeader,
+  ListPage,
+} from './factory';
+import {
+  Kebab,
+  navFactory,
+  ResourceKebab,
+  ResourceLink,
+  ResourceSummary,
+  SectionHeading,
+} from './utils';
+
+
+export const machineConfigReference = referenceForModel(MachineConfigModel);
+export const machineConfigMenuActions = Kebab.factory.common;
```

Comment:
Might want to make a copy of this to avoid ever accidentally mutating the original.

```suggestion
export const machineConfigMenuActions = [...Kebab.factory.common];
```